sdks/java/core/src/main/java/org/apache/beam/sdk/util/RowJson.java

```diff
@@ -515,7 +537,11 @@ private void writeValue(JsonGenerator gen, FieldType type, Object value) throws
           writeRow((Row) value, type.getRowSchema(), gen);
           break;
         case LOGICAL_TYPE:
-          writeValue(gen, type.getLogicalType().getBaseType(), value);
+          if (KNOWN_LOGICAL_TYPE_IDENTIFIERS.contains(type.getLogicalType().getIdentifier())) {
+            gen.writeString(value.toString()); // ISO 8601 format
```

Comment:
This happens to work for the current logical types but I think there is a risk it will subtly break if the underlying type isn't something that guarantees toString is ISO 8601. How about this pattern:

```
if (SqlTypes.DATE.getIdentifier().equals(type.getLogicalType().getIdentifier())) {
    gen.writeString(((LocalDate) value).toString()); // ISO 8601 format
}
```
(Can you add the cast to 'DATETIME' above as well?)

Reply:
Done.